### PR TITLE
[16.0][ADD] sale_stock_partner_warehouse

### PR DESCRIPTION
**Pull request by santostelmo**
### Description
Allow to choose by default a warehouse on SO based on a Partner parameter. 
If the warehouse parameter is completed, then this value would be the default one on the sale. 
If empty, then the warehouse on sale would be the standard default warehouse.

### CODE REVIEW

File: sale_stock_partner_warehouse/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "author": "Camptocamp, BCIM, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "data": [
+        "views/partner_view.xml",
+    ],
+    "depends": [
+        "sale_stock",
+        "sale_stock_warehouse_multicompany",
```
Comment (jbaudoux) [inline sale_stock_partner_warehouse/__manifest__.py]: You don't need this dependency
```suggestion
```

File: sale_stock_partner_warehouse/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    warehouse_id = fields.Many2one(
+        compute="_compute_warehouse_id", store=True, precompute=True
+    )
+
+    @api.depends("user_id", "company_id", "partner_id.sale_warehouse_id")
```
Comment (jbaudoux) [inline sale_stock_partner_warehouse/models/sale_order.py]: If you change the wh on the partner, you probably don't want to recompute all SO of that partner
```suggestion
    @api.depends("user_id", "company_id", "partner_id")
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]
This module is not related to multi-company and must move to sale-workflow repo

File: sale_stock_partner_warehouse/__manifest__.py (hunk 1 above)
Comment (santostelmo) [reply sale_stock_partner_warehouse/__manifest__.py]: Removed dependency

**Comment by santostelmo** [comment]
> This module is not related to multi-company and must move to sale-workflow repo

Moving it to sale-workflow: https://github.com/OCA/sale-workflow/pull/3044

**Comment by santostelmo** [comment]
@jbaudoux can you plase check the PR on OCA sale-workflow ?

**Pull request closed without merging**